process/block/preprocess/gasComputation.go

```diff
@@ -165,20 +170,28 @@ func (gc *gasComputation) ComputeGasConsumedByTx(
 		return 0, 0, process.ErrNilTransaction
 	}
 
-	txGasLimitConsumption := gc.economicsFee.ComputeGasLimit(txHandler)
-
-	if core.IsSmartContractAddress(txHandler.GetRcvAddr()) {
-		if txSenderShardId != txReceiverShardId && txGasLimitConsumption < txHandler.GetGasLimit() {
-			gasConsumedByTxInSenderShard := txGasLimitConsumption
-			gasConsumedByTxInReceiverShard := txHandler.GetGasLimit() - txGasLimitConsumption
+	moveBalanceConsumption := gc.economicsFee.ComputeGasLimit(txHandler)
+
+	txType := gc.txTypeHandler.ComputeTransactionType(txHandler)
+	isSCCall := txType == process.SCDeployment ||
+		txType == process.SCInvoking ||
+		txType == process.BuiltInFunctionCall ||
+		(core.IsSmartContractAddress(txHandler.GetRcvAddr()) && len(txHandler.GetData()) > 0)
```

Comment:
Isn't this process.SCInvoking ?

Reply:
no, in case of cross call it is MOVE balance.